test/fuzz/transformation_changing_memory_semantics_test.cpp

```diff
@@ -0,0 +1,482 @@
+               OpReturn
+               OpFunctionEnd
+  )";
+  const auto env = SPV_ENV_UNIVERSAL_1_3;
+  const auto consumer = nullptr;
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+  // Bad: Memory semantics can't change with the same value.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 0, 26)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: OpAtomicLoad memory semantics can't be a release value.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 0, 27)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: Can't change memory semantics higher bits value.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(22, SpvOpAtomicLoad, 0), 0, 30)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad:OpAtomicLoad memory semantics can't be a release value.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(22, SpvOpAtomicLoad, 0), 0, 27)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: OpAtomicLoad memory semantics can't be a release value. Old memory
+  // semantics value must be lower than new.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(23, SpvOpAtomicLoad, 0), 0, 29)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: OpAtomicLoad memory semantics can't be a release value.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(23, SpvOpAtomicLoad, 0), 0, 27)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: OpAtomicLoad memory semantics can't be a release value.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(23, SpvOpAtomicLoad, 0), 0, 250)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: OpAtomicLoad memory semantics can't be a release value.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(23, SpvOpAtomicLoad, 0), 0, 28)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Successful transformations.
+  {
+    // Changing memory semantics from |Acquire| to |SequentiallyConsistent| with
+    // keeping the old higher bits value.
```

Comment:
Good comments! I suggest changing it a bit:

```
// OpAtomicLoad Acquire                | UniformMemory
// to:
// OpAtomicLoad SequentiallyConsistent | UniformMemory
```

Reply:
Done.